### PR TITLE
feat: added thrift language server 0.2.1

### PR DESCRIPTION
**Pull request by ThanePatrol**
## Description of changes

Added new thrift language server. Addresses https://github.com/NixOS/nixpkgs/issues/340764
I use this package daily and have wrapped it myself. 
It is commonly use at companies that use Thrift for defining IDL files. 
I am keen to be a maintainer of this package

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by roberth** [comment]
This is actually great, because a very similar PR was opened minutes before you did, which means that you could both maintain it, and have a reviewer.

The only notable change is `doCheck = false;` in the other PR. I don't have much of an opinion about reusing `pname`, or SRI hashes.

Haven't seen this happen before, but just to make things easy for the both of you, I propose to close this PR and add `ThanePatrol` as the first maintainer in the list (in the other PR). I hope that makes it a fair outcome, but we could switch it around (switch the order of maintainers and close the other PR).

> keen to be a maintainer

Does that sound alright to you @ThanePatrol and @callumio?

I don't know either of you, so fill me in if there's anything I need to know.

**Comment by callumio** [comment]
> This is actually great, because [a very similar PR](https://github.com/NixOS/nixpkgs/pull/340767) was opened minutes before you did, which means that you could both maintain it, and have a reviewer.
> 
> The only notable change is `doCheck = false;` in the other PR. I don't have much of an opinion about reusing `pname`, or SRI hashes.
> 
> Haven't seen this happen before, but just to make things easy for the both of you, I propose to close this PR and add `ThanePatrol` as the first maintainer in the list (in the other PR). I hope that makes it a fair outcome, but we could switch it around (switch the order of maintainers and close the other PR).
> 
> > keen to be a maintainer
> 
> Does that sound alright to you @ThanePatrol and @callumio?
> 
> I don't know either of you, so fill me in if there's anything I need to know.

That seems fair to me - I'd be happy to share maintainer-ship with @ThanePatrol and add them as first maintainer in my PR - though they don't seem to be in the maintainers file or have a commit in this PR with maintainer details (please do point me in the correct direction if I'm mistaken).

**Comment by roberth** [comment]
Right, and that's why ofborg's red.
@ThanePatrol could you add yourself to the maintainers list? See
- https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
- https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md

**Comment by ThanePatrol** [comment]
@roberth Added myself to the maintainer list in this PR. Apologies for not updating the maintainer-list.nix earlier. 

@callumio Do you want to make the changes to add maintainership to your PR or use this one?

**Comment by roberth** [comment]
> Apologies

Don't worry about it.

> your PR or use this one?

Whichever you find easier, but regardless, could you squash the commits about the maintainers list?

**Comment by ThanePatrol** [comment]
Thanks @roberth Squashed commits

I believe it's easier to continue with this branch so the changes to maintainer list do not need to be done in @callumio branch

File: maintainers/maintainer-list.nix
Comment (callumio) [inline maintainers/maintainer-list.nix:1]: This is a lot of changes (looks like due to formatting) to add yourself to the maintainers file.
This should also be in a separate commit __before the thrift-ls commit__ titled `maintainers: add hughmandalidis`

File: pkgs/by-name/th/thrift-ls/package.nix
Comment (callumio) [inline pkgs/by-name/th/thrift-ls/package.nix:1]: Similarly this file should also be in a separate commit (after the maintainers list changes), titled `thrift-ls: init at 0.2.1`.

File: pkgs/by-name/th/thrift-ls/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  vendorHash = "sha256-YoZ2dku84065Ygh9XU6dOwmCkuwX0r8a0Oo8c1HPsS4=";
+
+  postInstall = ''
+    mv $out/bin/thrift-ls $out/bin/thriftls
+  '';
+
+  meta = with lib; {
+    description = "Language server for Thrift";
```
Comment (callumio) [inline pkgs/by-name/th/thrift-ls/package.nix:25]:
```suggestion
    description = "Thrift Language Servert";
```
This is more personal opinion than anything, however this follows the description on the source repo.

File: pkgs/by-name/th/thrift-ls/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "joyme123";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Qib/xbTQiR0VpHsCOt5HGkbytaQ0GSCqaHVYFNPYGVs=";
+  };
+
+  vendorHash = "sha256-YoZ2dku84065Ygh9XU6dOwmCkuwX0r8a0Oo8c1HPsS4=";
+
+  postInstall = ''
+    mv $out/bin/thrift-ls $out/bin/thriftls
```
Comment (callumio) [inline pkgs/by-name/th/thrift-ls/package.nix:21]: Could you explain why you're renaming the binary? I don't really have an issue with it, but wondering about the motivation.

File: pkgs/by-name/th/thrift-ls/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "joyme123";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Qib/xbTQiR0VpHsCOt5HGkbytaQ0GSCqaHVYFNPYGVs=";
+  };
+
+  vendorHash = "sha256-YoZ2dku84065Ygh9XU6dOwmCkuwX0r8a0Oo8c1HPsS4=";
+
```
Comment (callumio) [inline pkgs/by-name/th/thrift-ls/package.nix:19]: Consider adding linker flags to strip debug symbols

```suggestion

  ldflags = [
    "-s"
    "-w"
  ];

```

**Review by callumio — CHANGES_REQUESTED** [review]

**Comment by callumio** [comment]
> Thanks @roberth Squashed commits
> 
> I believe it's easier to continue with this branch so the changes to maintainer list do not need to be done in @callumio branch

As for continuing on this branch, that's fine by me :smile:.
Please follow [the commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) though and feel free to ask if you're unsure.

**Comment by Aleksanaa** [comment]
As https://github.com/NixOS/nixpkgs/pull/340767 has a better and cleaner condition, I preferred to merge that one. Sorry for your efforts!

File: pkgs/by-name/th/thrift-ls/package.nix (hunk 2 above)
Comment (ThanePatrol) [reply pkgs/by-name/th/thrift-ls/package.nix:21]: Renaming the binary to match behavior of upstream. 
If we don't change the name then the binary will work for users with nvim lsp-config

Comment (ThanePatrol) [reply pkgs/by-name/th/thrift-ls/package.nix:21]: I'll leave ownership to you though as I am not a maintainer

Comment (callumio) [reply pkgs/by-name/th/thrift-ls/package.nix:21]: Yep makes sense - go for it :smile:
